service/authorization/authorization.go

```diff
@@ -109,6 +110,15 @@ func NewRegistration() serviceregistry.Registration {
 				}
 			}
 
+			// Get Token Endpoint
+			oidcCfg, err := auth.DiscoverOIDCConfiguration(context.Background(), srp.OTDF.AuthN.GetIssuer(), srp.Logger)
+			if err != nil {
+				logger.Error("failed to discover OIDC configuration", slog.String("error", err.Error()))
+				panic(fmt.Errorf("failed to discover OIDC configuration: %w", err))
```

Comment:
I'd rather do this during a readiness/startup check. The smell here is using context.Background. The problem is this will cause panics during startup for transient oidc connectivity issues, e.g. if you start keycloak and platform at the same time, this will almost certainly panic at first. Instead, it should just report as not started/ not ready.

Reply:
To do this move all the logic into another helper method that is then passed into the `srp.RegisterReadinessFunc()`

Reply:
I'll file a follow up about moving logic to readiness checks and improving error conditions of service handlers

Reply:
Filed as https://github.com/opentdf/platform/issues/1165